core/src/edu/asu/c3simulator/simulation/business/Company.java

```diff
@@ -0,0 +1,374 @@
+package edu.asu.c3simulator.simulation.business;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Data object to store data related to, and simulate conduct by a Company. In this model,
+ * a company is made up of its employees, assets (including capital, credit, inventory,
+ * and other non-liquid assets), {@link ConsumerConnection}s, products, and activity (e.g.
+ * marketing campaigns, purchase orders, etc).
+ * <p>
+ * NOTE: {@link Company} assumes that instances of {@link Product} are unique, and specify
+ * a type of product rather than a single manufactured unit.
+ * 
+ * @author Moore, Zachary
+ *
+ */
+public class Company
+{
+	private String name;
+	private List<Employee> employees;
+	
+	/**
+	 * Products currently manufactured and sold by this company, paired with the
+	 * production information about them. If a product is sold but not manufactured, the
+	 * associated value will be null.
+	 */
+	private HashMap<Product, ProductionInformation> productLine;
+	
+	private HashMap<Product, Integer> inventory;
+	private HashMap<Product, Integer> salesInCurrentMonth;
+	private HashMap<Product, Float> retailPrice;
+	private Credit credit;
+	private List<MarketingCampaign> activeMarketing;
+	private List<ConsumerConnection> salesOutlets;
+	
+	private HashMap<Product, Integer> publicOpinionOfProducts;
+	private int publicOpinionOfCompany;
+	
+	/**
+	 * Simulates the sale of a product, by calculating profits and adding it to the
+	 * company's current capital.
+	 * <p>
+	 * This method is intended to be called by Simulation and its agents, as part of a
+	 * market/economy simulation.
+	 * 
+	 * @param product
+	 *            The product to be bought/sold
+	 * @param quantity
+	 *            The quantity to buy/sell
+	 * @return True if the sale was successful, false if the sale was unable to be
+	 *         processed. Note, a false return is indicitive of the quantity being too
+	 *         high for this company to support a sale. For instance, the quantity may
+	 *         cause the company to exceed its monthly sales potential, as specified by
+	 *         {@link #getMonthlySalePotential(Product)}
+	 */
+	public boolean registerSale(Product product, int quantity)
```

Comment:
Function registerSale has two returns. Would it be better to only have only one return for debugging purposes? Just curious.